test/e2e/environment.go

```diff
@@ -154,12 +169,21 @@ func CreateEnvironment(t *testing.T, ctx context.Context, opts ...TestEnvOption)
 	if len(opt.Image) == 0 {
 		opt.Image = getOperatorImage(t)
 	}
-	kustomizeDir := PrepareKustomizeDir(t, opt.Image)
+
+	var kustomizeDir KustomizeDir
+	if opt.InstallViaKustomize {
+		kustomizeDir = PrepareKustomizeDir(t, opt.Image)
+	}
 
 	env, err := builder.Build(ctx)
 	require.NoError(t, err)
+
 	t.Cleanup(func() {
-		cleanupEnvironment(t, context.Background(), env, kustomizeDir.Tests())
+		if opt.InstallViaKustomize {
+			cleanupEnvironment(t, context.Background(), env, kustomizeDir.Tests())
+		} else { //nolint:revive,staticcheck
+			// TODO: using helm for installation, don't clean up because we do not know the release name.
+		}
```

Comment:
```suggestion
		}
```
As I understand we decided to not deploy with helm in this helper, but make this responsibility of the upgrade test body itself, so it's no longer a todo here, right?

Reply:
Good point, 621149c3